spring-integration-zip/src/main/java/org/springframework/integration/zip/transformer/ZipTransformer.java

```diff
@@ -98,90 +98,103 @@ public void setUseFileAttributes(boolean useFileAttributes) {
+			lastModifiedDate = new Date();
+		}
 
-			java.util.List<ZipEntrySource> entries = new ArrayList<ZipEntrySource>();
+		java.util.List<ZipEntrySource> entries = new ArrayList<ZipEntrySource>();
 
-			if (payload instanceof Iterable<?>) {
-				int counter = 1;
+		if (payload instanceof Iterable<?>) {
+			int counter = 1;
 
-				String baseName = FilenameUtils.getBaseName(zipEntryName);
-				String fileExtension = FilenameUtils.getExtension(zipEntryName);
+			String baseName = FilenameUtils.getBaseName(zipEntryName);
+			String fileExtension = FilenameUtils.getExtension(zipEntryName);
 
-				if (StringUtils.hasText(fileExtension)) {
-					fileExtension = FilenameUtils.EXTENSION_SEPARATOR_STR + fileExtension;
-				}
+			if (StringUtils.hasText(fileExtension)) {
+				fileExtension = FilenameUtils.EXTENSION_SEPARATOR_STR + fileExtension;
+			}
 
-				for (Object item : (Iterable<?>) payload) {
+			for (Object item : (Iterable<?>) payload) {
 
-					final ZipEntrySource zipEntrySource = createZipEntrySource(item, lastModifiedDate, baseName + "_"
-							+ counter + fileExtension, this.useFileAttributes);
-					if (logger.isDebugEnabled()) {
-						logger.debug("ZipEntrySource path: '" + zipEntrySource.getPath() + "'");
-					}
-					entries.add(zipEntrySource);
-					counter++;
+				final ZipEntrySource zipEntrySource = createZipEntrySource(item, lastModifiedDate, baseName + "_"
+						+ counter + fileExtension, this.useFileAttributes);
+				if (logger.isDebugEnabled()) {
+					logger.debug("ZipEntrySource path: '" + zipEntrySource.getPath() + "'");
 				}
-			}
-			else {
-				final ZipEntrySource zipEntrySource =
-						createZipEntrySource(payload, lastModifiedDate, zipEntryName, this.useFileAttributes);
 				entries.add(zipEntrySource);
+				counter++;
 			}
+		}
+		else {
+			final ZipEntrySource zipEntrySource =
+					createZipEntrySource(payload, lastModifiedDate, zipEntryName, this.useFileAttributes);
+			entries.add(zipEntrySource);
+		}
 
-			final byte[] zippedBytes = SpringZipUtils.pack(entries, this.compressionLevel);
+		final byte[] zippedBytes = SpringZipUtils.pack(entries, this.compressionLevel);
 
-			if (ZipResultType.FILE.equals(this.zipResultType)) {
-				final File zippedFile = new File(this.workDirectory, zipFileName);
-				FileCopyUtils.copy(zippedBytes, zippedFile);
-				zippedData = zippedFile;
-			}
-			else if (ZipResultType.BYTE_ARRAY.equals(this.zipResultType)) {
-				zippedData = zippedBytes;
-			}
-			else {
-				throw new IllegalStateException("Unsupported zipResultType " + this.zipResultType);
-			}
+		if (ZipResultType.FILE.equals(this.zipResultType)) {
+			final File zippedFile = new File(this.workDirectory, zipFileName);
+			FileCopyUtils.copy(zippedBytes, zippedFile);
+			zippedData = zippedFile;
+		}
+		else if (ZipResultType.BYTE_ARRAY.equals(this.zipResultType)) {
+			zippedData = zippedBytes;
+		}
+		else {
+			throw new IllegalStateException("Unsupported zipResultType " + this.zipResultType);
+		}
 
+		try {
```

Comment:
Ha! Sure thing :smile: 
Will move it before Message building.
Just thought that it would be logical to care about resource in the final end.
But, eh, that's true: isn't good idea to delete in case of error.
Doh...